tests/common/include/tests/common/path_correction.inl

```diff
@@ -7,463 +7,171 @@
-    getter::element(bound_cov, e_bound_qoverp, e_bound_qoverp) = 1.f;
-    getter::element(bound_cov, e_bound_time, e_bound_time) = 1.f;
-
-    // bound track parameter
-    const bound_track_parameters<transform3> bound_param0(0u, bound_vector,
-                                                          bound_cov);
-
-    // Actors
-    target_aborter::state targeter{0u};
-    pathlimit_aborter::state pathlimit_state{path_limit};
-    propagation::print_inspector::state print_insp_state{};
-    parameter_transporter<transform3>::state bound_updater{};
-    parameter_resetter<transform3>::state rst{};
-
-    propagator_t p({}, {});
-    propagator_t::state propagation(bound_param0, det.get_bfield(), det);
-    crk_stepper_t::state &crk_state = propagation._stepping;
-
-    // The track is strongly bent, allow for overstepping
-    crk_state().set_overstep_tolerance(-1.f * unit<scalar>::um);
-
-    // Retrieve navigation information
-    auto &debug_printer = propagation._navigation.inspector();
-
-    // Run propagator
-    ASSERT_TRUE(p.propagate(
-        propagation, std::tie(targeter, pathlimit_state, print_insp_state,
-                              bound_updater, rst)));
-
-    // Make sure, the track was propagated to the end
-    ASSERT_NEAR(crk_state.path_length(), S, env::tol)
-        << print_insp_state.to_string() << debug_printer.to_string();
-    EXPECT_NEAR(crk_state().pos()[0], 0.f, env::tol);
-    EXPECT_NEAR(crk_state().pos()[1],
-                2.f * std::cos(constant<scalar>::pi / 6.f), env::tol);
-    EXPECT_NEAR(crk_state().pos()[2],
-                2.f * std::sin(constant<scalar>::pi / 6.f), env::tol);
-
-    // Bound state after one turn propagation
-    const auto bound_param1 = crk_state._bound_params;
-    const auto bound_vec0 = bound_param0.vector();
-    const auto bound_vec1 = bound_param1.vector();
-    const auto bound_cov0 = bound_param0.covariance();
-    const auto bound_cov1 = bound_param1.covariance();
-
-    // Check if the bound state stays the same after one turn propagation
-
-    // vector
-    for (unsigned int i = 0u; i < e_bound_size; i++) {
-        EXPECT_NEAR(matrix_operator().element(bound_vec0, i, 0u),
-                    matrix_operator().element(bound_vec1, i, 0u), env::tol);
-    }
+    // Prepare transform matrices
+    std::vector<transform3> trfs;
+    for (std::size_t i = 0; i < n_planes; i++) {
 
-    // covariance
-    for (unsigned int i = 0u; i < e_bound_size; i++) {
-        for (unsigned int j = 0u; j < e_bound_size; j++) {
-            EXPECT_NEAR(matrix_operator().element(bound_cov0, i, j),
-                        matrix_operator().element(bound_cov1, i, j), env::tol);
-        }
+        const scalar s = step_size * scalar(i);
+
+        // Translation of the new surface
+        const auto T = reference_helix(s);
+
+        // Normal vector of the new surface
+        auto w = reference_helix.dir(s);
+        w = axis_rot(w);
```

Comment:
So, while I am not against getting rid of the ```telescope_geometry```, the reason you write this surface placement code is that the surfaces need to be rotated away from the helix direction? Why is that?

Reply:
Nvm, otherwise there will be no path correction term necessary?

Reply:
In that case it seems to be re-doing what we already have? Just as an alternative:
```
const auto tel_det = create_telescope_detector(
        host_mr, rectangle, n_surfaces, tel_length, silicon_tml<scalar>(),
        80.f * unit<scalar>::um, reference_helix);
const auto& trfs = tel_det.trnasform_store();
```
You can also add an rotation after the fact. I'll leave it up to you

Reply:
> otherwise there will be no path correction term necessary?

It still needs it